### ISSUE TITLE
Hooks not installed in the right hooks directory for worktrees

### ISSUE THREAD

**Issue by s0undt3ch** [pre-commit/pre-commit#808]
When on a git worktree, the hooks are not installed on the right hook dir:
```
❯ pre-commit install
pre-commit installed at /home/vampas/projects/SaltStack/console/raas/.git/worktrees/revan1/hooks/pre-commit
```
The hook does not run.

The hook only runs when I moved the `pre-commit` script to `/home/vampas/projects/SaltStack/console/raas/.git/hooks/pre-commit`

```
❯ pre-commit --version
pre-commit 1.10.4
```

**Comment by asottile** [comment]
oh subtrees, the bane of my existence 😆. this doesn't surprise me since I don't use them myself but they should work so I'll look into a fix for this!

thanks for the report!

**Comment by s0undt3ch** [comment]
Thank you for looking into it

**Comment by asottile** [comment]
I'm able to reproduce!

The odd thing:

```
asottile@asottile-VirtualBox:/tmp/wat$ git rev-parse --git-dir
/tmp/pre-commit-hooks/.git/worktrees/wat
```

I would think that git hooks would be installable / runnable from there.  I almost suspect this is a bug in `git` itself.

**Comment by asottile** [comment]
Apparently, there's `git rev-parse --git-common-dir` which gives the correct value

**Comment by asottile** [comment]
however, that's new in `git` 2.5, so I'll need to try both

**Comment by asottile** [comment]
via #809

**Comment by brandon-leapyear** [comment]
Hi, I still have problems with this in a git worktree. I installed `pre-commit v1.18.0`:

```
$ echo $PWD
/Users/bchinn/leapyear/lybeta
$ git rev-parse --git-common-dir --git-dir
/Users/bchinn/leapyear/main/.git
/Users/bchinn/leapyear/main/.git/worktrees/lybeta
$ pre-commit install
...
pre-commit installed at /Users/bchinn/leapyear/main/.git/hooks/pre-commit
```

**Comment by asottile** [comment]
A worktree shares its `.git` directory with the original repo, so your output is correct as far as I can tell

**Comment by brandon-leapyear** [comment]
oh interesting. So a worktree doesn't share a `.git` directory; it's found at `/other/repo/.git/worktrees/thisrepo/` (this is where the usual git files are stored, like storing `FETCH_HEAD` and stuff), but apparently worktrees use the main git directory for git hooks. Thanks!

**Comment by asottile** [comment]
Yep yep! Work trees are weird

**Comment by DavidBAEpstein** [comment]
One could have a hard (or soft) link between pre-commit in .git/hooks
and another copy that lives inside the worktree. Then it can be developed with the usual branch/PR process. Any reason why this shouldn't work properly?

**Comment by asottile** [comment]
@DavidBAEpstein it could work that way except that's not how git works (and I have zero control over how git works!).  git worktrees share the `.git` directory with the base repository

**Comment by DavidBAEpstein** [comment]
@asottile Thanks. It's all much more complicated than I imagined, as I found when I tried to read about it on Google. Would my suggestion work in the simplest situation, with only one worktree? I was hoping to use git's pre-commit without downloading your pre-commit package. (Sorry!). I am very new to git and Github, and can't cope with all the complications. Everything I read seems to assume more knowledge than I have.
